### PR TITLE
Created the dadjoke trigger

### PR DESCRIPTION
**Pull request by parsec**
And here we have the dadjoke trigger. Not entirely sure I did this correctly... I tried to use Zerolog since I was doing a whole request thingy with net/http but uh...
idk I just copied what @m-242 did. Hopefully this works. We shall see... 

If not, welp, back to the drawing board. Tried to stay away from dependency hell on this one.

### CODE REVIEW

**Comment by parsec** [comment]
Still not sure why the docker build is failing.

**Comment by parsec** [comment]
> 
> 
> Still not sure why the docker build is failing.

ignore me, i'm dumb. it shows logs

File: dadjokes.go [hunk 1]
```diff
@@ -26,7 +27,7 @@ func httpRequest() string {
 	req, err := http.NewRequest("GET", url, nil) // crafting HTTP request with url and userAgent
 	if err != nil { // if err, loggging error via Zerolog and returning error to chat
 		sublogger.Warn().Err(err).Msg("HTTP request formatting error")
-		return "An error occurred: " + string(err)
+		return "An error occurred: " + string(err.Error())
```
Comment (m-242) [inline dadjokes.go:30]:
```suggestion
              return fmt.Sprintf("An error occured: %v", err)
```

Comment (m-242) [reply dadjokes.go:30]: err.Error() already returns a string, it's useless to cast it's return to a string.

File: dadjokes.go [hunk 2]
```diff
@@ -35,21 +36,21 @@ func httpRequest() string {
 	resp, err := client.Do(req)
 	if err != nil { 
 		sublogger.Warn().Err(err).Msg("HTTP request error")
-		return "An error occurred: " + string(err)
+		return "An error occurred: " + string(err.Error())
```
Comment (m-242) [inline dadjokes.go:39]: same

File: dadjokes.go [hunk 3]
```diff
@@ -35,21 +36,21 @@ func httpRequest() string {
...
-		sublogger.Warn().Err(errors.New).Msg("HTTP status code")
+		sublogger.Warn().Err(err).Msg("HTTP status code")
 		return "Error: HTTP Status Code " + statusCode
 	}
 
 	r, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		sublogger.Warn().Err(err).Msg("ioutil.ReadAll error")
-		return "An error occurred: " + string(err)
+		return "An error occurred: " + string(err.Error())
```
Comment (m-242) [inline dadjokes.go:53]: same

**Review by m-242 — CHANGES_REQUESTED** [review]

File: dadjokes.go (hunk 1 above)
Comment (parsec) [reply dadjokes.go:30]: Oh, I didn't know that!

Comment (parsec) [reply dadjokes.go]: addressed by 10243c2

File: dadjokes.go (hunk 2 above)
Comment (parsec) [reply dadjokes.go]: addressed by 10243c2

File: dadjokes.go (hunk 3 above)
Comment (parsec) [reply dadjokes.go]: addressed by 10243c2

**Pull request merged**